cadctl/cmd/credentials/credentials.go

```diff
@@ -0,0 +1,73 @@
+package credentials
+
+import (
+	"fmt"
+	"github.com/openshift/configuration-anomaly-detection/cadctl/cmd/investigate"
+	"github.com/openshift/configuration-anomaly-detection/pkg/services/assumerole"
+	"github.com/openshift/configuration-anomaly-detection/pkg/services/ccam"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+var Cmd = &cobra.Command{
+	Use:   "credentials",
+	Short: "Determines if a cluster's credentials are missing.",
```

Comment:
**suggestion:** I know this is a WIP, but we need to have a clear plan of the following to finish the implementation:
- how do we want to trigger this check? (cronJob?)
- how many clusters do we want to check per check? How many times does cadctl run per pipeline?  Keep in mind pipelines are a limited resource (we have a hard cap)
- what passes the cluster ids to check to this check? The cronJob? Does CAD query the clusters itself (might make CAD harder to test)?



My suggestion:
- cronJob triggers every 24 hours, queries all clusters (in supported versions?) from OCM and starts a pipeline for every 100 clusters it queried from OCM.
- the credential check command takes the tekton pipeline payload (the list of ~100 clusters) as input, and performs the check for all these clusters. Keep in mind that the cluster states may already have altered between the query and the credentials checks. Some might already be uninstalled or currently uninstalling.

Reply:
Hey Claudio, I'm in full agreement with your suggestion here. I'm out on vacation all next week but I'll look into some of this stuff when I get back.